pkg/mons/restore_quorum.go

```diff
@@ -0,0 +1,298 @@
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func RestoreQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace, goodMon string) {
+	err := restoreQuorum(context, operatorNamespace, clusterNamespace, goodMon)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func restoreQuorum(context *k8sutil.Context, operatorNamespace, clusterNamespace, goodMon string) error {
+	monCm, err := context.Clientset.CoreV1().ConfigMaps(clusterNamespace).Get(ctx.TODO(), MonConfigMap, v1.GetOptions{})
+	if err != nil {
+		log.Fatalf("failed to get mon configmap %s %v", MonConfigMap, err)
+	}
+
+	monData := monCm.Data["data"]
+	monEndpoints := strings.Split(monData, ",")
+
+	var goodMonPublicIp, goodMonPort string
+	var badMons []string
+	for _, m := range monEndpoints {
+		monName, monEndpoint, ok := strings.Cut(m, "=")
+		if !ok {
+			return fmt.Errorf("failed to fetch mon endpoint")
+		} else if monName == goodMon {
+			goodMonPublicIp, goodMonPort, ok = strings.Cut(monEndpoint, ":")
+			if !ok {
+				return fmt.Errorf("failed to get good mon endpoint and port")
+			}
+		} else {
+			badMons = append(badMons, monName)
+		}
+		fmt.Printf("mon=%s, endpoints=%s\n", monName, monEndpoint)
+
+	}
+
+	if goodMonPublicIp == "" {
+		return fmt.Errorf("error: good mon %s not found", goodMon)
+	}
+
+	fsidSecret, err := context.Clientset.CoreV1().Secrets(clusterNamespace).Get(ctx.TODO(), "rook-ceph-mon", v1.GetOptions{})
+	if err != nil {
+		log.Fatalf("failed to get mon configmap %s %v", MonConfigMap, err)
+	}
+
+	cephFsid := string(fsidSecret.Data["fsid"])
+	if cephFsid == "" {
+		return fmt.Errorf("ceph cluster fsid not found")
+	}
+
+	fmt.Printf("printing fsid secret %s\n", cephFsid)
+	fmt.Println("Check for the running toolbox")
+
+	_, err = debug.VerifyDeploymentExists(context, clusterNamespace, "rook-ceph-tools")
+	if err != nil {
+		return fmt.Errorf("failed to deployment rook-ceph-tools. %v", err)
+	}
+
+	toolBox, err := k8sutil.WaitForPodToRun(context, clusterNamespace, "app=rook-ceph-tools")
+	if err != nil || toolBox.Name == "" {
+		return fmt.Errorf("failed to get the running toolbox")
+	}
+
+	fmt.Printf("Restoring mon quorum to mon %s %s\n", goodMon, goodMonPublicIp)
+	fmt.Printf("The mons to discard are: %s\n", badMons)
+	fmt.Printf("The cluster fsid is %s\n", cephFsid)
+
+	var answer, output string
+	fmt.Printf("Are you sure you want to restore the quorum to mon %s? yes-really-restore\n", goodMon)
+	fmt.Scanf("%s", &answer)
+	output, err = promtToContinueOrCancel(answer)
+	if err != nil {
+		return fmt.Errorf(" restoring the mon quorum to mon %s cancelled", goodMon)
+	}
+	fmt.Println(output)
+
+	fmt.Println("Waiting for operator pods to stop")
+	err = debug.SetDeploymentScale(context, operatorNamespace, "rook-ceph-operator", 0)
+	if err != nil {
+		return fmt.Errorf("failed to stop deployment rook-ceph-operator. %v", err)
+	}
+	fmt.Println("deployment.apps/rook-ceph-operator scaled")
```

Comment:
```suggestion
	fmt.Println("rook-ceph-operator deployment scaled down")
```

Reply:
this is same log that we get today, you want to update?

Reply:
yes let's update this one too

Reply:
added